pkg/msr/msr_linux.go

```diff
@@ -183,9 +183,11 @@ func (m MSR) Write(c CPUs, data ...uint64) []error {
 	return nil
 }
 
-// TestAndSet takes a mask of bits to clear and to set, and applies them to the specified MSR in
-// each of the CPUs. Note that TestAndSet does not write if the mask does not change the MSR
-func (m MSR) TestAndSet(c CPUs, clearMask uint64, setMask uint64) []error {
+// testAndSetMaybe takes a mask of bits to clear and to set, and applies them to the specified MSR in
+// each of the CPUs. It will set the MSR only if the value is different and a set is requested.
+// If the MSR is different for any reason that is an error.
+func (m MSR) testAndSetMaybe(c CPUs, clearMask uint64, setMask uint64, set bool) []error {
```

Comment:
Perhaps it makes more sense to call this `testAndOptionallySet`?

Reply:
that's ok, but the linux kernel seems to be leaning to maybe for stuff like this?